File: sally/leetcode/190_Reverse Bits.cpp
// 190. Reverse Bits
// Runtime: 0 ms, faster than 100.00% of C++ online submissions for Reverse Bits.
// Memory Usage: 6 MB, less than 8.36% of C++ online submissions for Reverse Bits.

class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t res = 0;
        for(int i = 0; i < 32; i++)
            if(n & ((unsigned int)1 << i)) {
                res |= ((unsigned int)1 << (32-i-1));
            }
        return res;
    }
};

File: coodingpenguin/leetcode/190_ReverseBits.py
# 문제: 190. Reverse Bits
# 링크: https://leetcode.com/problems/reverse-bits/


# 시간/공간: 32ms / 14.3MB
class Solution:
    def reverseBits(self, n: int) -> int:
        return int("{:032b}".format(n)[::-1], 2)
